packages/formstr-app/src/containers/ResponsesNew/index.tsx

```diff
@@ -13,14 +13,30 @@ import { fetchKeys, getAllowedUsers, getFormSpec } from "../../utils/formUtils";
 import { Export } from "./Export";
 import { Field, Tag } from "../../nostr/types";
 import { useApplicationContext } from "../../hooks/useApplicationContext";
+import { getDefaultRelays } from "@formstr/sdk";
+import { ZapModal } from "./zapModal";
+import { ThunderboltFilled } from "@ant-design/icons";
+import { fetchProfiles, fetchZapReceipts, formatCompact, ProfileInfo, ZapInfo } from "../../utils/zapUtils";
 
 const { Text } = Typography;
 
+interface SelectedResponse {
+  pubkey: string;
+  lud16?: string;
+  name?: string;
+  eventId: string;
+}
+
 export const Response = () => {
   const [responses, setResponses] = useState<Event[] | undefined>(undefined);
   const [formEvent, setFormEvent] = useState<Event | undefined>(undefined);
   const [formSpec, setFormSpec] = useState<Tag[] | null | undefined>(undefined);
   const [editKey, setEditKey] = useState<string | undefined | null>();
+  const [profiles, setProfiles] = useState<Map<string, ProfileInfo>>(new Map());
```

Comment:
should we maintain a common profiles map across the App ? maybe in applicationContext

Reply:
Let's make it common, and it can store all profiles it fetched in a session as a cache, we shouldn't fetch the same profiles repeatedly